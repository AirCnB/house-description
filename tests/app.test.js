import React from 'react';
import App from '../client/component/app.jsx';


const wrapper = shallow(<App />);
const instance = wrapper.instance();

describe('<App />', () => {
  it('renders main App', () => {
  	expect(wrapper).toMatchSnapshot();
  });
  it('renders all components', () => {
  	expect(wrapper.children()).toHaveLength(7);
  })
})

describe('App states', () => {
  it('properly sets showModal state', () => {
  	expect(wrapper.state().showModal).toEqual(false);
    instance.togglePopup();
    expect(wrapper.state().showModal).toEqual(true);
  });
  it('properly sets showDescription state', () => {
  	expect(wrapper.state().showDescription).toEqual(false);
    instance.toggleDescription();
    expect(wrapper.state().showDescription).toEqual(true);
  });
  it('properly sets showRules state', () => {
  	expect(wrapper.state().showRules).toEqual(false);
    instance.toggleRules();
    expect(wrapper.state().showRules).toEqual(true);
  });
  it('properly sets showCancellation state', () => {
  	expect(wrapper.state().showCancellation).toEqual(false);
    instance.toggleCancellation();
    expect(wrapper.state().showCancellation).toEqual(true);
  });
});
	







