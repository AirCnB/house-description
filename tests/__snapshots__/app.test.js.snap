// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders main App 1`] = `
<div
  className="main-app"
>
  <DescriptionTitle
    city=""
    home_name=""
    host=""
    house_type=""
    images=""
    info={Object {}}
    titleIcons={Array []}
  />
  <HighlightList
    highlights={Array []}
    host=""
    info={Object {}}
  />
  <DescriptionBody
    description=""
    info={Object {}}
    showDescription={false}
    toggleDescription={[Function]}
  />
  <Amenities
    amenities={Array []}
    info={Object {}}
    modal={false}
    showModal={false}
    togglePopup={[Function]}
  />
  <SleepingArrangements
    info={Object {}}
    sleep_arrangement={Array []}
    titleIcon={Array []}
  />
  <Rules
    info={Object {}}
    rules={Array []}
    rules2=""
    showRules={false}
    toggleRules={[Function]}
  />
  <Cancellations
    showCancellation={false}
    toggleCancellation={[Function]}
  />
</div>
`;
